src/Bundle/Doctrine/ORM/ResourceRepositoryTrait.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ResourceBundle\Doctrine\ORM;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\QueryBuilder;
+use Pagerfanta\Adapter\ArrayAdapter;
+use Pagerfanta\Doctrine\ORM\QueryAdapter;
+use Pagerfanta\Pagerfanta;
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+/**
+ * @property EntityManagerInterface $_em
+ * @property ClassMetadata          $_class
+ *
+ * @method QueryBuilder createQueryBuilder(string $alias, string $indexBy = null)
+ * @method ?object      find($id, $lockMode = null, $lockVersion = null)
+ */
+trait ResourceRepositoryTrait
```

Comment:
What would be awesome, is to create php unit test for this trait with usage of anonymous class. But I can live with the current solution. Also, it appears that it may be cumbersome to do such a test